### PR TITLE
use guice 5.0.1

### PR DESCRIPTION
**Pull request by cdietrich**
[WIP ]use guice 5.0.1
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.dev-bom/build.gradle [hunk 1]
```diff
@@ -12,7 +12,7 @@ dependencies {
         api "com.google.guava:guava-gwt:30.1-jre"
         api "com.google.guava:guava-testlib:30.1-jre"
         api "com.google.guava:guava:30.1-jre"
-        api "com.google.inject:guice:3.0"
+        api "com.google.inject:guice:5.0.1"
```
Comment (cdietrich) [inline org.eclipse.xtext.dev-bom/build.gradle:15]: should we add aopalliance here too?

File: org.eclipse.xtext.dev-bom/build.gradle [hunk 2]
```diff
@@ -27,11 +27,11 @@ dependencies {
         api "org.eclipse.emf:org.eclipse.emf.ecore.change:2.14.0"
         api "org.eclipse.emf:org.eclipse.emf.ecore.xmi:2.16.0"
         api "org.eclipse.emf:org.eclipse.emf.ecore:2.20.0"
-        api "org.eclipse.emf:org.eclipse.emf.mwe.core:1.6.1"
-        api "org.eclipse.emf:org.eclipse.emf.mwe.utils:1.6.1"
-        api "org.eclipse.emf:org.eclipse.emf.mwe2.launch:2.12.1"
-        api "org.eclipse.emf:org.eclipse.emf.mwe2.lib:2.12.1"
-        api "org.eclipse.emf:org.eclipse.emf.mwe2.runtime:2.12.1"
+        api "org.eclipse.emf:org.eclipse.emf.mwe.core:[1.6.1,1.7.0)"
```
Comment (cdietrich) [inline org.eclipse.xtext.dev-bom/build.gradle:30]: am not sure if this is a good idea

File: org.eclipse.xtext.dev-bom/build.gradle (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.dev-bom/build.gradle:16]: i have added it

File: org.eclipse.xtext.dev-bom/build.gradle (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.dev-bom/build.gradle:31]: but it allows to use the beta in mwe without using fake milestone or snapshot here

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtext.dev-bom/build.gradle (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.dev-bom/build.gradle:16]: Maybe use a version range for Guice

Comment (cdietrich) [reply org.eclipse.xtext.dev-bom/build.gradle:16]: i fear version ranges

**Pull request merged**